internal/pkg/app/routingTableManager.go

```diff
@@ -108,36 +120,40 @@ func (r *DefaultRoutingTableManager) AddRoute(ctx context.Context, routeConfig *
 	if err != nil {
 		return err
 	}
-	// set up sender
-	sender, err := r.pluginMgr.RegisterSender(ctx, routeConfig.Sender.Plugin, routeConfig.Sender.Name, stringify(routeConfig.Sender.Config))
-	if err != nil {
-		return err
-	}
-	// set up receiver
-	receiver, err := r.pluginMgr.RegisterReceiver(ctx, routeConfig.Receiver.Plugin, routeConfig.Receiver.Name, stringify(routeConfig.Receiver.Config))
-	if err != nil {
-		return err
-	}
+	var lrw LiveRouteWrapper
 	// set up filter chain
-	filterChain := &filter.Chain{}
+	lrw.FilterChain = &filter.Chain{}
 	if routeConfig.FilterChain != nil {
 		for _, f := range routeConfig.FilterChain {
 			filter, err := r.pluginMgr.RegisterFilter(ctx, f.Plugin, f.Name, stringify(f.Config))
-			filterChain.Add(filter)
 			if err != nil {
+				lrw.Unregister(ctx, r)
 				return err
 			}
+			lrw.FilterChain.Add(filter)
 		}
 	}
+	// set up sender
+	lrw.Sender, err = r.pluginMgr.RegisterSender(ctx, routeConfig.Sender.Plugin, routeConfig.Sender.Name, stringify(routeConfig.Sender.Config))
+	if err != nil {
+		lrw.Unregister(ctx, r)
+		return err
+	}
+	// set up receiver
+	lrw.Receiver, err = r.pluginMgr.RegisterReceiver(ctx, routeConfig.Receiver.Plugin, routeConfig.Receiver.Name, stringify(routeConfig.Receiver.Config))
+	if err != nil {
+		lrw.Unregister(ctx, r)
+		return err
+	}
 	// create live route
-	liveRoute := &route.Route{}
+	lrw.Route = &route.Route{}
 	r.Lock()
-	r.liveRouteMap[routeConfig.Id] = &LiveRouteWrapper{liveRoute, sender, receiver, filterChain}
+	r.liveRouteMap[routeConfig.Id] = &lrw
 	r.Unlock()
 	go func() {
 		sctx := context.Background()
 		//TODO: use application context here? see issue #51
-		err = liveRoute.Run(sctx, receiver, filterChain, sender) // run is blocking
+		err = lrw.Route.Run(sctx, lrw.Receiver, lrw.FilterChain, lrw.Sender) // run is blocking
 		if err != nil {
 			log.Ctx(ctx).Error().Str("op", "AddRoute").Msg(err.Error())
```

Comment:
Do we need to unregister here?

Reply:
Interesting - if running the route fails we should probably also remove it from the routing table. Will think about that for a moment.

Reply:
Per discussion with Michael we will ignore runtime errors for now and only unregister (partial) routes during creation - what types of errors to expect from a running route and how to react to that in the routing manager is a separate discussion.